src/components/Chat/Chat.tsx

```diff
@@ -13,12 +13,19 @@ import QuickReplyList from '../QuickReplyList';
 
 export interface ChatProps {
   endPoint: string;
+  referralParameter?: string;
 }
 
-const Chat: (props: ChatProps) => JSX.Element = ({ endPoint }: ChatProps) => {
-  const { messages, quickReplies, sendMessage, sendQuickReply, sendAction }: UseTock = useTock(
+const Chat: (props: ChatProps) => JSX.Element = ({ endPoint, referralParameter }: ChatProps) => {
+  const { messages, quickReplies, sendMessage, sendQuickReply, sendAction, sendReferralParameter }: UseTock = useTock(
     endPoint
   );
+    if(referralParameter) {
+        useEffect(() => {
+            sendReferralParameter(referralParameter);
+
+        }, [sendReferralParameter]);
+    }
```

Comment:
```js
    useEffect(() => {
        if (referralParameter) {
            sendReferralParameter(referralParameter);
        }
    }, [sendReferralParameter, referralParameter]);
```

Better syntax (complies with the [rules of hooks](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level))

Reply:
Done